File: majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        int mx=0;
        int ans;
        for(auto x:mp){
            if(x.second>mx){
                mx=x.second;
                ans=x.first;
            }
        }
        return ans;
    }
};

File: majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 1;
        int majority = nums[0];
        for(int i = 1; i< nums.length; i++){
            if(majority == nums[i]){
                count++;
            }else{
                count--;
                if(count == 0){
                    majority = nums[i];
                    count = 1;
                }
            }
        }
        return majority;
    }
}